src/frame/header.rs

```diff
@@ -16,6 +16,11 @@ pub(crate) struct Header {
 impl Header {
     pub fn for_payload(frame_type: FrameType, payload: &[u8]) -> Header {
         assert!(payload.len() < crate::BLOCK_NUM_BYTES);
+        // TODO(trinity): I make the argument the frame_type should be checksummed too:
```

Comment:
You are absolutely right.

This format is loosely based on leveldb log format, which also checksums the type
https://github.com/google/leveldb/blob/main/doc/log_format.md